src/leo-client-app/src/components/logModal.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+
+
+type Props = {
+    open: boolean;
+    logData: any;
+    handleClose: () => void;
+  };
+
+export default function LogDialog({ open, logData, handleClose }: Props) {
+
+const downloadLogs = () => {
+  const file = new Blob([JSON.stringify(logData?.data?.message)], {type: 'text/plain'});
+  const url = URL.createObjectURL(file);
+  const link = document.createElement("a");
+  link.download = "logFile.txt";
+  link.href = url;
+  link.click();
+  handleClose();
+}
+
+  return (
+    <React.Fragment>
```

Comment:
why a react fragment and not a div/stack/whatever?

Reply:
just a dialog now

Reply:
u didnt answer the question lmfao

Reply:
I am experimenting stuff. If I get smthing to work, I can optimize it after